docs/reduction-in-locks.md

```diff
@@ -0,0 +1,106 @@
+
+### Benefits
+
+The `--lock-ddl=REDUCED` option key features are as follows:
+
+* Rather than holding the backup lock throughout the entire backup process, the lock is only acquired for a very short period.
+
+* DDL operations are allowed on the server while the backup is in progress.
+
+* The redo log tracks all file operations to maintain consistency. The backup process creates new metadata files to ensure data remains intact. When you run `--prepare`, the system first processes these metadata files to create a consistent database state.
+
+### Limitations
+
+* Disk read and write operations are performed during the execution of `DDL` operations and while backing up data using `xtrabackup`. Both types of operations involve accessing and modifying the data stored on disk, which can be resource-intensive.
+
+* Backups may be larger because new `DDL` operations are executed concurrently on the server, and the files they generate are included in the backup. Additionally, certain `DDL` operations, such as `ADD INDEX` or encryption changes to existing data files, will cause these data files to be recopied, further increasing the backup size.
+
+* The number of open file handles in your operating system should be configured to match the number of files in the server data directory.
+
+## How the backup lock with the `--lock-ddl` option works
+
+Percona Server for MySQL implemented `LOCK TABLES FOR BACKUP` to block `DDL` operations on a server. Percona `xtrabackup` utilizes this lock during the backup process to ensure that any `DDL` operation does not corrupt the backup. With MySQL-compatible servers `xtrabackup` utilizes `Lock Instance For Backup` to lock the server during the backup. The lock does not affect the `Data Manipulation Language` (DML) operations.
+
+With the [`--lock-ddl=ON`](./xtrabackup-option-reference.md/#lock-ddl) option the backup lock is enabled by default from the start of the backup process.
+{ .power-number }
+
+1. Set a backup lock on the server to prevent any DDL operations, such as creating or altering tables.
+
+2. Parse and copy the redo logs from the checkpoint up to the current `Log Sequence Number` (LSN), and start tracking new file operations.
+
+3. Identify and copy the `.ibd` files.
+
+4. Copy `non-InnoDB` files.
+
+5. Gather a sync point from all engines (InnoDB LSN, binlog, Global Transaction Identifier (GTID), etc.) by querying the log status.
+
+6. Stop the redo thread once it copies at least up to sync point.
+
+7. Release the backup lock on the server.
+
+If the backup lock is disabled with the [`--lock-ddl=OFF`](./xtrabackup-option-reference.md/#lock-ddl) option, a backup continues while concurrent DDL operations are executed. These backups may be invalid and may fail at either the `backup` or the `--prepare` step.
+
```

Comment:
This entire section should be in how xtrabackup works. And it is not important to describe here. 

May be somewhere we can point to the 'how lock-ddl=on works'  from this page

Reply:
Agree. This section should be moved with a link in the next section.

Reply:
I moved this section to the option reference doc.